src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx

```diff
@@ -0,0 +1,103 @@
+
+## Requirements [#requirements]
+
+To collect logs from your containers, you need to have:
+
+    * Infrastructure agent [1.8.32](/docs/release-notes/infrastructure-release-notes/infrastructure-agent-release-notes/new-relic-infrastructure-agent-1832) or higher running on Linux
+    * CentOS version 6.0 or higher, if you're using CentOS
+    * Docker with engine from [v1.12](https://docs.docker.com/engine/release-notes/prior-releases/#1120-2016-07-28) or other [`containerd`-based container runtimes](https://kubernetes.io/docs/setup/production-environment/container-runtimes)
+
+## Collect logs from your containers [#collect-logs]
+
+1. Open the configuration file for the infrastructure agent and add this snippet:
+
+    ```yml
+    - name: containers
+      file: /var/lib/docker/containers/*/*.log
+    ```
+
+<img
+    title="Add the containers path to your snippet"
+    alt="A screenshot that shows where to add this snippet in context of your yaml file"
+    src={infrastructureAddContainersLogPath}
+/>
+
+* You may find this file at this path: `/etc/newrelic-infra/logging.d/logging.yml`.
+* This section directs the agent to collect logs from your containers
+
+2. Add a tag section to your Docker configuration that passes in the default `{{.Name}}` attribute:
+
+* Docker run command example
+
+    ```shell
+    docker run --log-opt tag="{{.Name}}" -d log-generator
+    ```
+
+* Docker Compose example in `docker-compose.yml`
+
+    ```yml
+    version: '3.9'
+    x-default-logging: &logging
+      driver: "json-file"
+      options:
+        max-size: "5m"
+        max-file: "2"
+        tag: "{{.Name}}"
+    ```
+
+<img
+    title="Configuration file in context"
+    alt="Screenshot that shows this snippet in context of a configuration file"
+    src={infrastructureAddNamesLogs}
+/>
+
+This adds a `attrs.tag` attribute to your logs, which looks like this snippet:
+
+    ```json
+        {
+        agentName:Infrastructure
+        agentVersion:1.50.0
+        **attrs.tag:cart-service**
+        coreCount:1
+        ...
+        }
+    ```
+
+3. Once the `attrs.tag` appears in your host-level logs, you can enable container logs by going to <DoNotTranslate>**[one.newrelic.com](https://one.newrelic.com) > Infrastructure </DoNotTranslate>, then click <DoNotTranslate>**HERE**</DoNotTranslate>.
+
+4. Create a log parsing rule by applying these configurations:
+
+    * Name: container name
+    * Field to parse: attrs.tag
+    * Filter logs based on NRQL: attrs.tag is not null
+    * Parsing rule: %{NOTSPACE:containerName}
+
+Your modal should look like this:
+
+    <img
+        title="Log parsing rules"
+        alt="Screenshot of log parsing in UI"
+        src={logsParsingUi}
+    />
+    <figcaption>
+
+    </figcaption>
+
+5. Generate some data, wait a few minutes, and you'll see container logs assigned to your containers running on your host, rather than assigned to the host itself.
```

Comment:
I'm confused here. The Infra-Agent log forwarder normally gets the `entity.guid` belonging to the _host_ running these containers, so the logs get associated to the host. Are you doing something differently in this case, when running Docker containers? My understanding is that by having `containerName` in the logs, you are somehow leveraging some Entity Synthesis rule to associate the log to a particular container. And I assume this is happening because the Infra-Agent actually sends `entity.guid.INFRA`, and not `entity.guid`. So by the time entity synthesis runs, `entity.guid` is still not populated.

I'm afraid that this solution is making some assumptions in the order in which the backend will perform the actions (that is, first it will execute parsing, then entity synthesis, and later on entity.guid consolidation), so I'd probably advise against this. I'd not like to have this coupling in the pipeline, as it may block us from being able to evolve it and modify the step order if we ever need to.

PS: have you verified that if logs get associated to the container they don't get associated to the host _as well_?